### PR TITLE
PUD-164: Add page numbering. Validate number of pages generated by gotenberg

### PR DESCRIPTION
**Pull request by philsegal**
Currently we are not validating that the numbers have been added since the bytes seem to vary with each execution and so cant be verified. Potentially we can look for an alternative way to validate.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/PdfDecorator.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import com.lowagie.text.pdf.PdfReader
+import com.lowagie.text.pdf.PdfStamper
+import org.librepdf.openpdf.fonts.Liberation
+import org.springframework.stereotype.Component
+import java.io.ByteArrayOutputStream
+
+@Component
+class PdfDecorator {
+
+  fun numberPages(pdfContent: ByteArray): ByteArray {
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/PdfDecorator.kt:15]: Given we have no unit test of this code how about the ugly solution of a comment like e.g.:
  // Note: this function is tested within CreateDossierComponentTest

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/DossierServiceTest.kt [hunk 2]
```diff
@@ -20,8 +21,9 @@ internal class DossierServiceTest {
   private val revocationOrderService = mockk<RevocationOrderService>()
   private val pdfDocumentGenerator = mockk<PdfDocumentGenerator>()
   private val recallDocumentService = mockk<RecallDocumentService>()
+  private val pdfDecorator = mockk<PdfDecorator>()
 
-  private val underTest = DossierService(revocationOrderService, pdfDocumentGenerator, recallDocumentService)
+  private val underTest = DossierService(revocationOrderService, pdfDocumentGenerator, recallDocumentService, pdfDecorator)
 
   @Test
   fun `get dossier returns part A, license and revocation order as dossier when all present for recall`() {
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/DossierServiceTest.kt:29]: You'll love this: add "as numbered pages" to test title?

**Review by AndyNewton — APPROVED** [review]
Approved, but I wouldn't mind a couple of comments added - given the absence of a unit test for the numberer.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/PdfDecorator.kt (hunk 1 above)
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/PdfDecorator.kt:15]: But its not tested :) its mocked in that test

**Review by mr-ken — APPROVED** [review]
Looks good

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/DossierServiceTest.kt (hunk 2 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/DossierServiceTest.kt:29]: we're not numbering it tho, its mocked.

Comment (AndyNewton) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/DossierServiceTest.kt:29]: Yeah, but the same is true of every step in this test.  IMO, it's a pretty weak test.  And yes, I wrote the original version.  The component test comes to the rescue fortunately.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/PdfDecorator.kt (hunk 1 above)
Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/PdfDecorator.kt:15]: Where's :mr-hankey: when you need him?

Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/PdfDecorator.kt:15]: OK.  I've added a unit test for `fun numberPages` now: https://github.com/ministryofjustice/manage-recalls-api/pull/165

**Pull request merged**